src/modules/commander/ModeUtil/mode_requirements.cpp

```diff
@@ -158,9 +158,13 @@ void getModeRequirements(uint8_t vehicle_type, failsafe_flags_s &flags)
 	// NAVIGATION_STATE_AUTO_TAKEOFF
 	setRequirement(vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF, flags.mode_req_angular_velocity);
 	setRequirement(vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF, flags.mode_req_attitude);
-	setRequirement(vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF, flags.mode_req_local_position);
```

Comment:
@sfuhrer How does it work with altitude, we actually need gobal altitude but only require a local one. I assume it's because we always have baro and assume that global alt is always available, right?

Reply:
I'll change it to only use local position in the takeoff flight mode.

Reply:
Leaving it for now. In https://github.com/PX4/PX4-Autopilot/pull/25123 I would have switched to local position but it didn't solve it for all, as TECS still requires global position.